enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py

```diff
@@ -164,23 +164,26 @@ def _process_work_order_by_id(self, wo_id):
                 logger.error("Received empty work order corresponding " +
                              "to id %s from wo-requests table", wo_id)
                 self._kv_helper.remove("wo-processing", wo_id)
+                self._kv_helper.remove("wo-worker-map",
+                                       self._enclave_id_hex)
                 return
 
         except Exception as e:
             logger.error("Problem while reading the work order %s "
                          "from wo-requests table", wo_id)
             self._kv_helper.remove("wo-processing", wo_id)
+            # Remove from wo-worker-map
+            # @TODO : Add to wo-worker-pending if the error occurred is
+            # recoverable, i.e.-retrying again could succeed.
+            self._kv_helper.remove("wo-worker-map",
+                                   self._enclave_id_hex)
             return
```

Comment:
Return value of _process_work_order_by_id() is inconsistent. In case of error it is returning without value and on success it is returning response. And caller side we are returning as it is.